trl/trainer/dpo_trainer.py

```diff
@@ -522,13 +649,51 @@ def make_inputs_require_grad(module, input, output):
         # Compute that only on the main process for faster data processing.
         # see: https://github.com/huggingface/trl/pull/1255
         with PartialState().local_main_process_first():
-            # tokenize the dataset, lower writer batch size to avoid OOM (frequent in vision models)
-            train_dataset = train_dataset.map(self.tokenize_row, num_proc=self.dataset_num_proc, writer_batch_size=10)
+            # Tokenize the datasets
+            train_dataset = train_dataset.map(
+                _tokenize,
+                fn_kwargs={
+                    "tokenizer": self.processor if self.is_vision_model else self.tokenizer,
+                    "is_vision_model": self.is_vision_model,
+                },
+                batched=True,
+                num_proc=self.dataset_num_proc,
+                remove_columns=train_dataset.column_names,
```

Comment:
I would keep `writer_batch_size=10`. Otherwise, it uses a batch size of 1000, which, when using images, can cause more than 2GB to be written at once, and raise an exception.
Furthermore, in my experience, lowering this value doesn't make the code any slower.